src/core/writer_csv.py

```diff
@@ -46,45 +47,35 @@ def add_track(self, track):
         """ Append track to database """
         name = os.path.join(self.dir, track['Track'])
         trkFile = open('%s.track' % name, "wb", **open_extra)
-        trkWriter = csv.DictWriter(trkFile, self.track_keys)
+        trkWriter = csv.DictWriter(trkFile, self.track_keys,extrasaction='ignore')
         trkWriter.writeheader()
         trkWriter.writerow(track)
         trkFile.close()
         
         self.lapFile = open('%s.laps' % name, "wb", **open_extra)
-        self.lapWriter = csv.DictWriter(self.lapFile, self.lap_keys)
+        self.lapWriter = csv.DictWriter(self.lapFile, self.lap_keys,extrasaction='ignore')
         self.lapWriter.writeheader()
 
-    def add_lap(self, lap):
-        """ Append lap to a database """
-        self.lapWriter.writerow(lap)
-
-    def begin_points(self, summary):
-        """ Points for track will follow """
-        track = summary['Track']
-        if self.lapFile:
-            self.lapFile.close()
-            self.lapFile = None
-        self.track = track
-        name = os.path.join(self.dir, track)
         self.ptsFile = open('%s.points' % name, "wb", **open_extra)
         self.ptsWriter = csv.DictWriter(self.ptsFile, self.point_keys)
         self.ptsWriter.writeheader()
 
+        for lap in track['LapData']:
+          self.add_lap(lap)
+
+        self.ptsFile.close()
+        self.lapFile.close()
+
+    def add_lap(self, lap):
+        """ Append lap to a database """
+        self.lapWriter.writerow(lap)
+        for point in lap['PointData']:
+          self.add_point(point)
+          
     def add_point(self, point):
         """ Append point to a database """
         self.ptsWriter.writerow(point)
 
-    def commit(self):
```

Comment:
Hmm, yeah can't remember why I got rid of this.  Probably just because I thought we didn't need it anymore with the way I had rewritten the code.